blocks/identity-block/components/FormInputField/index.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+
+import { PrimaryFont } from '@wpmedia/shared-styles';
+
+import './styles.scss';
+
+export const FIELD_TYPES = {
+  EMAIL: 'email',
+  TEXT: 'text',
+  PASSWORD: 'password',
+};
+
+const FormInputField = ({
+  defaultValue = '',
+  label,
+  name,
+  isDefaultError = false,
+  onChange = () => {},
+  placeholder,
+  minLength = 0,
+  maxLength = 0,
+  required = false,
+  tip,
+  type = FIELD_TYPES.TEXT,
+  validationErrorMessage,
+  validationPattern,
+}) => {
+  const [error, setError] = useState(isDefaultError);
+  const [errorMessage, setErrorMessage] = useState(validationErrorMessage);
+
+  const handleBlur = (event) => {
+    setError(!!event.target.validationMessage);
+    setErrorMessage(event.target.validationMessage
+      ? validationErrorMessage || event.target.validationMessage
+      : '');
+  };
+
+  const handleChange = (event) => {
```

Comment:
Preference on this.  I didn't want the field to be updating error states while you were typing, but I thought it was alright to clear the error when the user completes the "puzzle".

I'm definitely open to debate on the functionality on this.  It could even change in the future when we get design input.